app/models/concerns/provideable.rb

```diff
@@ -8,13 +8,17 @@ module Provideable
   included do
     belongs_to :organization # Automatically validates presence as of Rails 5
 
-    validates :contact_name, presence: { message: "Must provide a name or a business name" }, if: proc { |ddp| ddp.business_name.blank? }
-    validates :business_name, presence: { message: "Must provide a name or a business name" }, if: proc { |ddp| ddp.contact_name.blank? }
+    validates :contact_name, presence: { message: "Must provide a name or a business name" }, if: proc { |ddp| ddp.business_name.blank? && validation_required? }
+    validates :business_name, presence: { message: "Must provide a name or a business name" }, if: proc { |ddp| ddp.contact_name.blank? && validation_required? }
```

Comment:
So this seems a bit odd. The main issue is that Provideable is only a mixin for a grand total of two classes - Vendor and ProductDriveParticipant.

If there's logic that is _not_ shared between these two classes, it should be taken out of the mixin entirely and added only to the class where it's relevant. In this case, since Vendor and ProductDriveParticipant no longer have the same business rules, these validations should be moved to ProductDriveParticipant, and we should have separate rules for Vendor.

Reply:
Ah this makes perfect sense, thank you!